webui/src/components/ToolList.jsx

```diff
@@ -189,18 +189,30 @@ class ToolCard extends React.Component {
     };
 
     renderHeader(imgSrc, tool, invocationURL) {
-        const trackCall = () => _paq.push(['trackEvent', 'Tools', 'StartTool', tool.name]);
+        const stopBubbling = (e) => {
+            e.stopPropagation();
+        }
+        const trackCall = (e) => {
+            stopBubbling(e);
+            _paq.push(['trackEvent', 'Tools', 'StartTool', tool.name]);
+        }
         const Highlighter = this.props.highlighter;
         return (
             <dl className="dl-horizontal header">
-                <dt><img src={imgSrc}/></dt>
+                <dt>
+                    <Indicator title={"chevron-" + (this.state.showDetails ? "down":"right")}
```

Comment:
is there a reason not use css classes and define styles for classes like elsewhere?

Reply:
I usually prefer inline styles for smaller elements, because it's clear what objects they apply to. The separation of style in a css file makes it more difficult to change the code, as one needs to find out what elements the style applies to. And most of the time the style is changed together with the element, so the change has to be done in two places. In the css file I put the styles for larger sections, transitions and media queries. I'm not very strict about it, though.

Reply:
Ok, now I understand there's (kind of) a rationale behind it. My concern would be that in many places you would using the same property values e.g. colours and it could be a pain to make overall changes in the style. Maybe a bit hypothetical issue so wouldn't worry about it too much.

Reply:
I do think it makes the JSX a bit harder to read but again not really a big deal

Reply:
I'm not sure I understand, can you show me how you would rewrite it?

Reply:
Sorry, this remark is not about these specific lines. I mean attributes like these:

```
... style={{fontSize:"75%", marginRight:0, color:'#444'}}>
```

```
<a style={{fontSize: 20}} onClick={stopBubbling} href={tool.homepage} target="_blank">
                        <Indicator title={"link"} style={{marginLeft:4, fontSize:"75%"}}></Indicator>
```

with a 'stand-off' CSS approach you would have a `className` attribute with a simple string value instead of `style` with often a multi-property object.

Reply:
Ok, yes, it's true that the jsx doesn't look pretty. What I could do is lift the inline style in a separate variable: 

````
const style = {
    fontSize:"75%", 
    marginRight:0, 
    color:'#444'
};
... style={style}>
````